src/ore/src/ssh_key.rs

```diff
@@ -0,0 +1,123 @@
+// Copyright Materialize, Inc. and contributors. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License in the LICENSE file at the
+// root of this repository, or online at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Utilities for generating and managing SSH keys.
+
+use openssl::pkey::{PKey, Private};
+use serde::{Deserialize, Serialize};
+use ssh_key::private::{Ed25519Keypair, Ed25519PrivateKey, KeypairData};
+use ssh_key::public::Ed25519PublicKey;
+use ssh_key::{LineEnding, PrivateKey};
+
+/// Keeps track of a public-private keypair, stored in the OpenSSH format
+#[derive(Debug, Eq, PartialEq, PartialOrd, Ord, Clone, Serialize, Deserialize)]
+pub struct SshKeypair {
+    public_key: Vec<u8>,
+    private_key: Vec<u8>,
```

Comment:
this is generally considered unsafe right? there are crates to have buffers like this that are zero'd when they are dropped (like https://crates.io/crates/secstr), but that might just be a followup

Reply:
I had a note to explore using `zeroize` (which is what `ssh-key` is using), but left it as a follow-up to allow us to test this flow sooner. Do you have any preferences/suggestions between `zeroize`/`secstr`?

Reply:
zeroize i have heard more about so id use `ssh-key` if possible